### ISSUE TITLE
Class decorators put before export

### ISSUE THREAD

**Issue by knownasilya** [ember-codemods/ember-native-class-codemod#53]
```
/app/components/toggle-archived/component.js
  5:1  error  Parsing error: Using the export keyword between a decorator and a class is not allowed. Please use `export @dec class` instead.

  3 |
  4 | @tagName('')
> 5 | export default class ToggleArchivedComponent extends Component {}
```

**Comment by knownasilya** [comment]
@pzuraq would that mean this is not compatible with ember-decorators v5 with the above codemod/flag?

**Comment by pzuraq** [comment]
@knownasilya we were waiting on TC39 to make the final call on whether these should be before or after export. The current default in `ember-decorators@v5` is decorators-before, and so you need to add the `eslint` flag (because `eslint` switched and doesn't read babel configs). This is customizable via flags to `@ember-decorators/babel-transforms` if you want to switch now: https://github.com/ember-decorators/ember-decorators/blob/master/packages/babel-transforms/index.js#L14

@ssutar we should actually update the codemod to put class decorators _after_ exports, TC39 has made the decision 😕

**Comment by ssutar** [comment]
@pzuraq @knownasilya 

I tried to change the parser options to set `decoratorsBeforeExport` to `false` as per https://babeljs.io/docs/en/babel-plugin-proposal-decorators#decoratorsbeforeexport However I still get the decorators before the export statements. Updating the `jscodeshift` version to latest did not work as well.

Basically the AST is same whether the decorators are before export or after so I think it should be as easy as config option but haven't been able to find it. Will keep looking for it, but if you know anything about it please let me know

**Comment by ssutar** [comment]
Logged an issue with jscodeshift: https://github.com/facebook/jscodeshift/issues/304

**Comment by rwjblue** [comment]
we can still fix, but it requires annoying string munging

https://github.com/scalvert/ember-es6-class-codemod/blob/e489ee64a8cda1bbe144e05c618427f104bcbbe7/transforms/ember-object/index.js#L21

the output is "just a string" here, the world is your oyster...

**Comment by pzuraq** [comment]
This is no longer relevant since the stage 2 decorators transform never took off, and we're still using stage 1.
